src/odemis/dataio/tiff.py

```diff
@@ -1955,4 +1718,461 @@ def read_thumbnail(filename):
+        desc = tiff_file.GetField(T.TIFFTAG_IMAGEDESCRIPTION)
+
+        if (desc and ((desc.startswith("<?xml") and "<ome " in desc.lower())
+                    or desc[:4].lower() == '<ome')):
+            try:
+                # take care of multiple file distribution
+
+                file_data, data = data, [] # keep the original data in case it doesn't go smoothly
+                path, basename = os.path.split(filename)
+                desc = re.sub('xmlns="http://www.openmicroscopy.org/Schemas/OME/....-.."',
+                            "", desc, count=1)
+                desc = re.sub('xmlns="http://www.openmicroscopy.org/Schemas/ROI/....-.."',
+                            "", desc)
+                root = ET.fromstring(desc)
+
+                # Keep track of the files that were already opened
+                file_read = set()
+    #             ifd_counter = 0
+                for tiff_data in root.findall("Image/Pixels/TiffData"):
+
+                    uuid = tiff_data.find("UUID")
+                    if uuid is None:
+                        # uuid attribute is only part of multiple files distribution
+                        continue
+                    else:
+                        uuid_data = uuid.get("FileName")
+    #                 tiff_data.set("IFD", ifd_counter)
+                    ifd_data = tiff_data.get("IFD")
+    #                 ifd_counter += 1
+                    # attach to the right path
+                    uuid_path = os.path.join(path, uuid_data)
+                    if uuid_data in file_read:
+                        continue
+                    # try to find and open the enlisted file
+                    try:
+                        f_link = TIFF.open(uuid_path, mode='r')
+                        self.opened_tiff_files.append(f_link)
+                    except TypeError:
+                        logging.warning("File '%s' enlisted in the OME-XML header is missing.", uuid_path)
+                        continue
+
+                    for dir_index in AcquisitionDataTIFF._iterDirectories(f_link):
+                        AcquisitionDataTIFF._createDataArrayShadows(f_link, dir_index, data, thumbnails)
+
+                    file_read.add(uuid_data)
+
+                # If this file was not enlisted in the xml data we assume it has
+                # been renamed. In this case we also include its data.
+                if basename not in file_read:
+                    data.extend(file_data)
+
+                data = AcquisitionDataTIFF._reconstructFromOMETIFF(desc, data, os.path.basename(filename))
+            except Exception:
+                # fallback to pretend there was no OME XML
+                logging.exception("Failed to decode OME XML string: '%s'", desc)
+
+        # Remove all the None (=thumbnails) from the list
+        content = [i for i in data if i is not None]
+
+        AcquisitionData.__init__(self, tuple(content), tuple(thumbnails))
+
+    def close(self):
```

Comment:
Let's try as hard as possible to get it working automatically when it's named `__del__`.